sdcm/utils/version_utils.py

```diff
@@ -1,68 +1,150 @@
+SCYLLA_VERSION_RE = re.compile(r"\d+(\.\d+)?\.[\d\w]+([.~][\d\w]+)?")
+PRIMARY_XML_GZ_REGEX = re.compile(r'="(.*?primary.xml.gz)"')
+
+REPOMD_XML_PATH = "repodata/repomd.xml"
+SCYLLA_URL_REGEX = re.compile(".*(?P<url>http.*scylladb.com.*).*")
+
+SCYLLA_URL_RESPONSE_TIMEOUT = 30
+SUPPORTED_XML_EXTENSIONS = ("xml", "xml.gz")
+SUPPORTED_FILE_EXTENSIONS = ("list", "repo", "Packages", "gz") + SUPPORTED_XML_EXTENSIONS
+VERSION_NOT_FOUND_ERROR = "The URL not supported, only Debian and Yum are supported"
+
+
+class ScyllaFileType(Enum):
+    DEBIAN = auto()
+    YUM = auto()
+
+
+FILE_REGEX_DICT = {
+    ScyllaFileType.DEBIAN: [
+        (re.compile(r"deb\s+\[arch=(?P<arch>.*?)\]\s(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$"),
+         "{url}/dists/{version_code_name}/{component}/binary-{arch}/Packages"),
+    ],
+    ScyllaFileType.YUM:
+        [
+            (re.compile(r"baseurl=(?P<url>http.*scylladb.com.*)"), "{url}" + REPOMD_XML_PATH),
+            (re.compile(r"deb\s+(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$"),
+             "{url}/dists/{version_code_name}/{component}/binary-amd64/Packages.gz"),
+    ],
+}
+
+# The variable "type" indices the type of URL (Debian or Yum)
+# The variable "urls" contains all urls using the Scylla pattern
+RepositoryDetails = namedtuple("RepositoryDetails", ["type", "urls"])
+
+
+def get_url_content(url, return_url_data=True):
+    response = requests.get(url=url)
+    if response.status_code != 200:
+        raise ValueError(f"The following repository URL '{url}' is incorrect")
+    response_data = response.text
+    if not response_data:
+        raise ValueError(f"The repository URL '{url}' not contains any content")
+    if return_url_data:
+        return response_data.split('\n')
+    # To overcome on Pylint's "inconsistent-return-statements", a value must be returned    return []
+    return []
+
+
+def get_scylla_urls_from_repository(repo_details):
+    urls = set()
+    for url in repo_details.urls:
+        for (url_regex, url_format) in FILE_REGEX_DICT[repo_details.type]:
+            match = url_regex.match(url)
+            if not match:
+                # Continue until find the correct Regex
+                continue
+
+            full_url = url_format.format(**match.groupdict())
+            # for scylla-manager we never used a noarch key
+            basearch_list = ["x86_64"] if 'scylla-manager' in full_url else ["x86_64", "noarch"]
+            for basearch in basearch_list:
+                urls.add(Template(full_url).substitute(basearch=basearch, releasever='7'))
+            # We found the correct regex and we can continue to next URL
+            break
+
+    urls = list(urls)
+    ParallelObject(objects=urls, timeout=SCYLLA_URL_RESPONSE_TIMEOUT).run(func=lambda _url: get_url_content(
+        url=_url, return_url_data=False))
+    return urls
+
+
+def get_branch_version_from_debian_repository(urls):
```

Comment:
why one function is called `get_branch_version_from_debian_repository` and the other `validate_yum_repository` ?
they are both doing the same thing, 
also pick one deb/rpm, repo/list, debian/centos, don't mix them since it's confusing

Reply:
Thanks!